File: code/graph_algorithms/src/data_structures/adjacency_list.py
class graph:
            def __init__(self):
                        self.graph={}
            def graph_input(self,n):
                        for i in range(1,n+1):

                                    self.graph.update({i:None})
                                    element=[ ]
                                    choice='y'
                                    while choice=='y':
                                                x=int(input("Enter the edge for vertex %d: "%i))
                                                element.append(x)
                                                self.graph[i]=element
                                                choice=input("Do you want to add more edges? (y/n):")
                        print("The adjacency list is:")
                        print(self.graph)

                
n=int(input("enter total number of nodes: "))
g=graph()
g.graph_input(n)
"""
Output:
enter total number of nodes: 3
Enter the edge for vertex 1: 2
Do you want to add more edges? (y/n):y
Enter the edge for vertex 1: 3
Do you want to add more edges? (y/n):n
Enter the edge for vertex 2: 1
Do you want to add more edges? (y/n):y
Enter the edge for vertex 2: 3
Do you want to add more edges? (y/n):n
Enter the edge for vertex 3: 1
Do you want to add more edges? (y/n):y
Enter the edge for vertex 3: 2
Do you want to add more edges? (y/n):n
The adjacency list is:
{1: [2, 3], 2: [1, 3], 3: [1, 2]}
"""

File: code/graph_algorithms/src/data_structures/adjacency_list.cpp
#include <iostream>
#include <vector>
using namespace std;

//Used for printing out the adjList after you finish building it
void printList(vector<vector<int>> adjList)
{
	cout << "The adjacency list is: " << endl;
	for (int i = 0; i < adjList.size(); i++)
	{
		cout << "Vertex " << i + 1 << ": ";
		for (int j = 0; j < adjList[i].size(); j++)
		{
			cout << adjList[i][j] << " ";
		}
		cout << endl;
	}
}

int main()
{
	int nodeNum;
	int numInput;
	char charInput;
	cout << "Input the number of nodes in the graph:" << endl;
	cin >> nodeNum;
	vector<vector<int>> adjList(nodeNum);
	for (int i = 0; i < nodeNum; i++)
	{
		bool moveOn = false;
		while (!moveOn)
		{
			cout << "Enter an edge for vertex " << i + 1 << ": " << endl;
			cin >> numInput;
			adjList[i].push_back(numInput);
			cout << "Do you want to add more edges? (y/n): " << endl;
			cin >> charInput;
			if (charInput == 'n')
				moveOn = true;
		}
	}
	printList(adjList);
	return 0;
}